### PR TITLE
Updated DAO Models

### PR DESCRIPTION
**Pull request by picksitquick**
@aprajshekhar please review

## Changes

  - Updated Models to mapManyToMany
  - made minor refactorings

### CODE REVIEW

File: src/main/java/com/gamedoora/model/dao/Skills.java [hunk 1]
```diff
@@ -30,14 +32,17 @@ public class Skills extends Audit implements Serializable {
 	@Column(name = "skill_description")
 	private String description;
 
-	@OneToMany(mappedBy = "skills")
+	@ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+	@JoinTable(name = "user_skills", joinColumns = @JoinColumn(name = "user_id"), inverseJoinColumns = @JoinColumn(name = "skill_id"))
 	Set<UserSkills> userSkills;
```
Comment (aprajshekhar) [inline src/main/java/com/gamedoora/model/dao/Skills.java:37]: Shouldn't this be `Set<Users>`

File: src/main/java/com/gamedoora/model/dao/Skills.java [hunk 2]
```diff
@@ -30,14 +32,17 @@ public class Skills extends Audit implements Serializable {
...
-	Set<SkillsSource> skillsSources;
+	@ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+	@JoinTable(name = "skills_sources", joinColumns = @JoinColumn(name = "skill_id"), inverseJoinColumns = @JoinColumn(name = "source_id"))
+	Set<Sources> skillsSources;
 
-	@OneToMany(mappedBy = "skills")
-	Set<RoleSkills> roleSkills;
+	@ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+	@JoinTable(name = "roles_skills", joinColumns = @JoinColumn(name = "role_id"), inverseJoinColumns = @JoinColumn(name = "skill_id"))
+	Set<Skills> skills;
```
Comment (aprajshekhar) [inline src/main/java/com/gamedoora/model/dao/Skills.java:45]: Shouldn't this be `Set<Roles>`?

File: src/main/java/com/gamedoora/model/dao/Sources.java [hunk 3]
```diff
@@ -24,8 +25,9 @@ public class Sources extends Audit implements Serializable {
 	@Column(name = "source_name")
 	private String name;
 
-	@OneToMany(mappedBy = "sources")
-	Set<SkillsSource> skillsSources;
+	@ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+	@JoinTable(name = "skills_sources", joinColumns = @JoinColumn(name = "skill_id"), inverseJoinColumns = @JoinColumn(name = "source_id"))
+	Set<Sources> sources;
```
Comment (aprajshekhar) [inline src/main/java/com/gamedoora/model/dao/Sources.java:30]: Shouldn't this be `Set<Skills>`?

File: src/main/java/com/gamedoora/model/dao/Tenants.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import javax.persistence.Table;
+
+@Getter
+@Setter
+@Builder
+@Entity
+@AllArgsConstructor
+@NoArgsConstructor
+@Table(name = "tenants")
+public class Tenants extends Audit {
```
Comment (aprajshekhar) [inline src/main/java/com/gamedoora/model/dao/Tenants.java:22]: Observational: Wouldn't there be a ManyToMany with Users?

**Review by aprajshekhar — CHANGES_REQUESTED** [review]
Thanks for the PR. Few queries. 

Also change `columnDefinition = "false"`  to `columnDefinition = "nullable=false"` wherever appropriate.

**Review by aprajshekhar — APPROVED** [review]
LGTM. There are merge conflicts. Please have a look at them.

**Comment by picksitquick** [comment]
Conflicts Resolved, we can now merge this branch.

**Pull request merged**